### PR TITLE
Add GPIO support for MSP432.

### PR DESCRIPTION
**Pull request by Misaka0x2730**
Difference from LM4F:
1. Added full-range pin drive strength (2, 4, 6, 8, 10, 12 mA).
2. Added AF defines for all pins. 
3. And many more things...

### CODE REVIEW

File: include/libopencm3/msp432/e4/gpio.h [hunk 1]
```diff
@@ -0,0 +1,2524 @@
...
+/** GPIO Alternate Function 8 Identifier */
+#define GPIO_AF8            0x8
+/** GPIO Alternate Function 11 Identifier */
+#define GPIO_AF11           0xB
+/** GPIO Alternate Function 13 Identifier */
+#define GPIO_AF13           0xD
+/** GPIO Alternate Function 14 Identifier */
+#define GPIO_AF14           0xE
+/** GPIO Alternate Function 15 Identifier */
+#define GPIO_AF15           0xF
```
Comment (karlp) [inline include/libopencm3/msp432/e4/gpio.h]: Can we just drop these? the af number is just ... the af number.

**Comment by karlp** [comment]
This all looks really good, can it just be in one commmit? are there any portions you want to land separately?

File: include/libopencm3/msp432/e4/gpio.h (hunk 1 above)
Comment (Misaka0x2730) [reply include/libopencm3/msp432/e4/gpio.h]: Yes, but i think it makes it easier to read.

**Comment by Misaka0x2730** [comment]
Okay, no problem.

**Pull request closed without merging**